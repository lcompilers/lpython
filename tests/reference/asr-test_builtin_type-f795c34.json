{
    "basename": "asr-test_builtin_type-f795c34",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/test_builtin_type.py",
    "infile_hash": "81fc4815e7adc95e05982852652f2cb26658ae4e46d7b635d0d5f580",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test_builtin_type-f795c34.stdout",
    "stdout_hash": "6e813b30a83bfc41b0ca67e40451b1b81edd29b56029334eb7bba89e",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}