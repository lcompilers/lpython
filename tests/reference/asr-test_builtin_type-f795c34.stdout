(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [test_builtin_type]
                                    []
                                    [(SubroutineCall
                                        2 test_builtin_type
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            test_builtin_type:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            CONST_LIST:
                                                (Variable
                                                    3
                                                    CONST_LIST
                                                    []
                                                    Local
                                                    (ListConstant
                                                        [(RealConstant
                                                            12.220000
                                                            (Real 8)
                                                        )
                                                        (RealConstant
                                                            14.630000
                                                            (Real 8)
                                                        )
                                                        (RealConstant
                                                            33.820000
                                                            (Real 8)
                                                        )
                                                        (RealConstant
                                                            19.180000
                                                            (Real 8)
                                                        )]
                                                        (List
                                                            (Real 8)
                                                        )
                                                    )
                                                    (ListConstant
                                                        [(RealConstant
                                                            12.220000
                                                            (Real 8)
                                                        )
                                                        (RealConstant
                                                            14.630000
                                                            (Real 8)
                                                        )
                                                        (RealConstant
                                                            33.820000
                                                            (Real 8)
                                                        )
                                                        (RealConstant
                                                            19.180000
                                                            (Real 8)
                                                        )]
                                                        (List
                                                            (Real 8)
                                                        )
                                                    )
                                                    Parameter
                                                    (Const
                                                        (List
                                                            (Real 8)
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            d:
                                                (Variable
                                                    3
                                                    d
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Dict
                                                        (Character 1 -2 ())
                                                        (Integer 4)
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    3
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            l:
                                                (Variable
                                                    3
                                                    l
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (List
                                                        (Integer 4)
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            s:
                                                (Variable
                                                    3
                                                    s
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Character 1 -2 ())
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    test_builtin_type
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(=
                                        (Var 3 i)
                                        (IntegerConstant 42 (Integer 4))
                                        ()
                                    )
                                    (=
                                        (Var 3 s)
                                        (StringConstant
                                            "Hello, LPython!"
                                            (Character 1 15 ())
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 3 l)
                                        (ListConstant
                                            [(IntegerConstant 1 (Integer 4))
                                            (IntegerConstant 2 (Integer 4))
                                            (IntegerConstant 3 (Integer 4))
                                            (IntegerConstant 4 (Integer 4))
                                            (IntegerConstant 5 (Integer 4))]
                                            (List
                                                (Integer 4)
                                            )
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 3 d)
                                        (DictConstant
                                            [(StringConstant
                                                "a"
                                                (Character 1 1 ())
                                            )
                                            (StringConstant
                                                "b"
                                                (Character 1 1 ())
                                            )
                                            (StringConstant
                                                "c"
                                                (Character 1 1 ())
                                            )]
                                            [(IntegerConstant 1 (Integer 4))
                                            (IntegerConstant 2 (Integer 4))
                                            (IntegerConstant 3 (Integer 4))]
                                            (Dict
                                                (Character 1 1 ())
                                                (Integer 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (IntrinsicScalarFunction
                                                ObjectType
                                                [(Var 3 i)]
                                                0
                                                (Character 1 12 ())
                                                (StringConstant
                                                    "<type 'i32'>"
                                                    (Character 1 12 ())
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "<type 'i32'>"
                                                (Character 1 12 ())
                                            )
                                            (Logical 4)
                                            (LogicalConstant
                                                .true.
                                                (Logical 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (IntrinsicScalarFunction
                                                ObjectType
                                                [(Var 3 s)]
                                                0
                                                (Character 1 12 ())
                                                (StringConstant
                                                    "<type 'str'>"
                                                    (Character 1 12 ())
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "<type 'str'>"
                                                (Character 1 12 ())
                                            )
                                            (Logical 4)
                                            (LogicalConstant
                                                .true.
                                                (Logical 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (IntrinsicScalarFunction
                                                ObjectType
                                                [(Var 3 l)]
                                                0
                                                (Character 1 18 ())
                                                (StringConstant
                                                    "<type 'list[i32]'>"
                                                    (Character 1 18 ())
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "<type 'list[i32]'>"
                                                (Character 1 18 ())
                                            )
                                            (Logical 4)
                                            (LogicalConstant
                                                .true.
                                                (Logical 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (IntrinsicScalarFunction
                                                ObjectType
                                                [(Var 3 d)]
                                                0
                                                (Character 1 23 ())
                                                (StringConstant
                                                    "<type 'dict[str, i32]'>"
                                                    (Character 1 23 ())
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "<type 'dict[str, i32]'>"
                                                (Character 1 23 ())
                                            )
                                            (Logical 4)
                                            (LogicalConstant
                                                .true.
                                                (Logical 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (IntrinsicScalarFunction
                                                ObjectType
                                                [(Var 3 CONST_LIST)]
                                                0
                                                (Character 1 25 ())
                                                (StringConstant
                                                    "<type 'Const[list[f64]]'>"
                                                    (Character 1 25 ())
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "<type 'Const[list[f64]]'>"
                                                (Character 1 25 ())
                                            )
                                            (Logical 4)
                                            (LogicalConstant
                                                .true.
                                                (Logical 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (IntrinsicScalarFunction
                                                ObjectType
                                                [(IntrinsicScalarFunction
                                                    ObjectType
                                                    [(Var 3 i)]
                                                    0
                                                    (Character 1 12 ())
                                                    (StringConstant
                                                        "<type 'i32'>"
                                                        (Character 1 12 ())
                                                    )
                                                )]
                                                0
                                                (Character 1 13 ())
                                                (StringConstant
                                                    "<type 'type'>"
                                                    (Character 1 13 ())
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "<type 'type'>"
                                                (Character 1 13 ())
                                            )
                                            (Logical 4)
                                            (LogicalConstant
                                                .true.
                                                (Logical 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (StringCompare
                                            (IntrinsicScalarFunction
                                                ObjectType
                                                [(IntrinsicScalarFunction
                                                    ObjectType
                                                    [(Var 3 CONST_LIST)]
                                                    0
                                                    (Character 1 25 ())
                                                    (StringConstant
                                                        "<type 'Const[list[f64]]'>"
                                                        (Character 1 25 ())
                                                    )
                                                )]
                                                0
                                                (Character 1 13 ())
                                                (StringConstant
                                                    "<type 'type'>"
                                                    (Character 1 13 ())
                                                )
                                            )
                                            Eq
                                            (StringConstant
                                                "<type 'type'>"
                                                (Character 1 13 ())
                                            )
                                            (Logical 4)
                                            (LogicalConstant
                                                .true.
                                                (Logical 4)
                                            )
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        5
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    5
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        5 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                )
        })
    []
)
